plugins/modules/cloud_provider_account.py

```diff
@@ -0,0 +1,201 @@
+    type: str
+  verification_signature:
+    description:
+      - Cloud VM instance identity signature
+      - Only needed to verify a cloud provider account for use with RHSM auto registration
+    type: str
+author:
+    - Patrick Easters (@patrickeasters)
+'''
+
+EXAMPLES = '''
+- name: Create a cloud provider account
+  redhatinsights.subscriptions.cloud_provider_account:
+    provider: AWS
+    id: 123456789012
+    nickname: my pretty neat aws account
+    refresh_token: aHR0cHM6Ly93d3cueW91dHViZS5jb20vd2F0Y2g/dj1kUXc0dzlXZ1hjUQo=
+
+- name: Delete a provider account
+  redhatinsights.subscriptions.cloud_provider_account:
+    provider: MSAZ
+    id: c11d7e74-adb4-4ce3-9a07-93c7a6f88cdf
+    state: absent
+    refresh_token: aHR0cHM6Ly93d3cueW91dHViZS5jb20vd2F0Y2g/dj1kUXc0dzlXZ1hjUQo=
+
+- name: Verify an account for use with auto registration
+  redhatinsights.subscriptions.cloud_provider_account:
+    provider: AWS
+    id: 123456789012
+    verification_identity: aHR0cHM6Ly93d3cueW91dHViZS5jb20vd2F0Y2g/dj1XYWFBTmxsOGgxOAo=
+    verification_signature: aHR0cHM6Ly93d3cueW91dHViZS5jb20vd2F0Y2g/dj1pblhDX2xhYi0zNAo=
+    refresh_token: aHR0cHM6Ly93d3cueW91dHViZS5jb20vd2F0Y2g/dj1kUXc0dzlXZ1hjUQo=
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.redhatinsights.subscriptions.plugins.module_utils.rhapi import RedHatAPIClient
+import base64
+
+def return_changed(module):
+  result = dict(changed=True)
+  module.exit_json(**result)
+
+
+def run_module():
+    module_args = dict(
+        state=dict(choices=['present', 'absent'], default='present'),
+        refresh_token=dict(type='str', required=True),
+        refresh_token_client=dict(type='str', default='rhsm-api'),
+        provider=dict(type='str', required=True),
+        id=dict(type='str', required=True),
+        nickname=dict(type='str', required=False, default=''),
+        verification_identity=dict(type='str', required=False, default=''),
+        verification_signature=dict(type='str', required=False, default=''),
+    )
+
+    result = dict(
+        changed=False,
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    client = RedHatAPIClient(module)
+
+    # Get existing cloud providers
+    enabled = client.get('/cloud_access_providers/enabled')
+    if enabled.status_code != 200:
+            module.fail_json(msg="Failed to retrieve provider accounts")
+    providers = enabled.json()
+
+    # Find existing account in list of enabled providers
+    account = {}
+    for p in providers['body']:
+        if p['shortName'] == module.params['provider']:
+            for a in p['accounts']:
+                if a['id'] == module.params['id']:
+                    account = a
+                    break
+            break
+    
+    # Delete account if needed
+    if account and module.params['provider'] == 'present':
```

Comment:
How does this work? `provider` is one of `['AWS', 'MSAZ', 'GCP']`, right?

Reply:
Yea, that's what I get for getting distracted while copy/pasting. 🙃